airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -434,7 +509,7 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             # We cap the number of workers to avoid hitting the Stripe rate limit
             # The limit can be removed or increased once we have proper rate limiting
             concurrency_level = min(config.get("num_workers", 2), _MAX_CONCURRENCY)
-            streams[0].logger.info(f"Using concurrent cdk with concurrency level {concurrency_level}")
-            return [StreamFacade.create_from_stream(stream, self, entrypoint_logger, concurrency_level) for stream in streams]
-        else:
-            return streams
+            main_streams[0].logger.info(f"Using concurrent cdk with concurrency level {concurrency_level}")
+            main_streams = [self._create_concurrent_stream(base_stream, concurrency_level) for base_stream in main_streams]
```

Comment:
we should also wrap the substreams with `StreamFacade.create_from_stream`

It's made a very big difference on large datasets:
~30 minutes vs 1:20
<img width="1199" alt="Screenshot 2023-10-26 at 10 28 50 PM" src="https://github.com/airbytehq/airbyte/assets/1451943/3fc475ec-f035-4ce2-aed9-52d30df8b234">

I ran the test with `StreamFacade.create_from_stream` [from this branch for reference ](https://github.com/airbytehq/airbyte/tree/alex/stripe_partitions)